v2/metrics/datadog.go

```diff
@@ -0,0 +1,132 @@
+package metrics
+
+import (
+	"context"
+	"log"
+	"net/url"
+	"time"
+
+	"github.com/DataDog/datadog-api-client-go/api/v1/datadog"
+	"github.com/DataDog/datadog-go/statsd"
+)
+
+// DatadogClient emits metrics directly to Datadog.
+type DatadogClient struct {
+	StatsdClient
```

Comment:
> I'm trying to understand the motivation for embedding StatsdClient here rather than putting the methods that implement it right on the DatadogClient

My intent was for `DatadogClient` and `StatsdClient` to be decoupled in the eyes of the caller:
* `metrics.Client`  is the interface I'll implement against
* `metrics.StatsdClient` is an implementation that I can get from `metrics.NewStatsdClient()`
* `metrics.DatadogClient` is an implementation that I can get from `metrics.NewDatadogClient()`
* a new `logd` transport might be implemented by `metrics.LogdClient` via `metrics.NewLogdClient()`.

As a user, you should care: `DatadogClient isA Client`, `StatsdClient isA Client`. The fact that `DatadogClient isA StatsdClient` is just an implementation detail - it was the fastest way to implement a `metrics.Client` with a statsd-ish protocol.

Some ideas for improvements:
* maybe those important relationships are worth stating with type assertions: (e.g. `var _ Client = (*DatadogClient)(nil)` ?
* maybe it's worth extracting an internal `statsdClient` struct for `StatsdClient` and `DatadogClient` to both embed? That would make the type hierarchy more clear.

WDYT?